add patches for PPC

vchuravy opened the pull request:
Fixes #38859, #38893, #38895, and #38896

depends on JuliaPackaging/Yggdrasil#2354

vchuravy committed:
add patches for PPC

wal004 commented:
> depends on [JuliaPackaging/Yggdrasil#2354](https://github.com/JuliaPackaging/Yggdrasil/pull/2354)

Would you be up for automatically managing dependent PRs? I built [Dpulls](https://github.com/marketplace/dpulls) to handle that - could be helpful if you often deal with dependent PRs - totally free for open source.

imciner2 committed:
Backport LLVM patch to fix LLVM build on GCC 11

GCC 11 has changed header dependencies again, so the build of LLVM
fails because <limits> wasn't being included when it should be. This
is already fixed upstream in LLVM 12.

vchuravy committed:
move LLVM assert subversion to llvm.mk

vchuravy committed:
update LLVM binaries

vchuravy committed:
use BINARYBUILDER_LLVM_ASSERTS for downloading assert builds

vchuravy commented on contrib/refresh_checksums.mk:
```diff
@@ -39,7 +39,7 @@ endef
 # It will also generate a `FOO_BB_TRIPLET=$(2)` make flag.
 define make_flags
-USE_BINARYBUILDER=$(if $(filter src,$(2)),0,1) $(call makevar,$(1))_BB_TRIPLET=$(if $(filter src,$(2)),,$(2)) LLVM_ASSERTIONS=$(if $(filter assert,$(3)),1,0) DEPS_GIT=0
+USE_BINARYBUILDER=$(if $(filter src,$(2)),0,1) $(call makevar,$(1))_BB_TRIPLET=$(if $(filter src,$(2)),,$(2)) BINARYBUILDER_LLVM_ASSERTS=$(if $(filter assert,$(3)),1,0) DEPS_GIT=0
```
Will be changed back in https://github.com/JuliaLang/julia/pull/38775
